### PR TITLE
Mac sqlite flag

### PR DESCRIPTION
**Pull request by mbmcgarry**
Fixes nosetests so that they can be run on Mac - all tests pass or skip.   (Des not address hdf5backtest unit tests).  Skips one test (smbchk) that doesn't work on Darwin.

### CODE REVIEW

File: tests/helper.py [hunk 1]
```diff
@@ -26,11 +28,26 @@ def clean_outs():
...
+    """Uses sqlite if platform is Mac, otherwise uses hdf5
     """
-    return all([t in db.root for t in tables])
+    return h5out if platform.system() == 'Linux' else sqliteout
 
+def tables_exist(outfile, tables):
+    """Checks if output database contains the specified tables.
+    """
+    if outfile == h5out:
+        db = tables.open_file(outfile, mode = "r")
```
Comment (gidden) [inline tests/helper.py]: this should use a context manager like `with tables.open_file() as f:` or close the file explicitly

File: tests/helper.py [hunk 2]
```diff
@@ -26,11 +28,26 @@ def clean_outs():
...
+        return all([t in db.root for t in tables])
+    else:
+        tables = [t.replace('/','') for t in tables]
```
Comment (gidden) [inline tests/helper.py]: space between comma

File: tests/helper.py [hunk 3]
```diff
@@ -47,6 +64,14 @@ def find_ids(data, data_table, id_table):
             ids.append(id_table[i])
     return ids
 
+
+def to_ary(outfile, a, k):
```
Comment (gidden) [inline tests/helper.py]: the api should probably be `to_ary(a, k)` and check `which_outfile()` in the conditional statement

File: tests/helper.py [hunk 4]
```diff
@@ -59,48 +84,71 @@ def exit_times(agent_id, exit_table):
 
     return exit_times
 
-def agent_time_series(f, names):
+def agent_time_series(outfile, names):
```
Comment (gidden) [inline tests/helper.py]: same api change here. query `which_outfile()` in the conditional

File: tests/helper.py [hunk 5]
```diff
@@ -59,48 +84,71 @@ def exit_times(agent_id, exit_table):
...
-
-    # Get specific tables and columns
-    agent_entry = f.get_node("/AgentEntry")[:]
-    agent_exit = f.get_node("/AgentExit")[:] if hasattr(f.root, 'AgentExit') \
-        else None
-
-    # Find agent ids
-    agent_ids = agent_entry["AgentId"]
-    agent_type = agent_entry["Prototype"]
+    print("second outfile is",outfile)
```
Comment (gidden) [inline tests/helper.py]: probably shouldn't have a print statement here

File: tests/helper.py [hunk 6]
```diff
@@ -59,48 +84,71 @@ def exit_times(agent_id, exit_table):
...
     for name, ids in agent_ids.items():
         for id in ids:
-            idx = np.where(agent_entry['AgentId'] == id)[0][0]
+            idx = np.where(to_ary(outfile, agent_entry,'AgentId') == id)[0]
             entries[name][agent_entry[idx]['EnterTime']] += 1
 
     # cumulative entries
     entries = {k: [sum(v[:i+1]) for i in range(len(v))] \
-                   for k, v in entries.items()}
+               for k, v in entries.items()}
```
Comment (gidden) [inline tests/helper.py]: i think this indentation is off

**Comment by mbmcgarry** [comment]
ping - finally passes CI!

**Comment by gidden** [comment]
thanks @mbmcgarry!

**Pull request merged**